### PR TITLE
[wip]

### PR DESCRIPTION
**Pull request by notslang**
I want to be able to use parse-torrent for converting torrent files to/from JSON so I can easily load them into a database like couchdb, edit/normalize them with jq (or similar tools), and create torrent files from database entries. So I've started doing some cleanup:
- remove redundant fields & copies of buffers left in the parsed object
- improve the cli so it can do encoding and decoding via stdio (without needing a filesystem)
- add support for some extra fields like `encoding` & `md5sum`
- ensure that information isn't lost when converting to/from JSON (especially anything that would change the infohash)

anyway, this is wip right now, but feel free to comment on the parts I've pushed so far

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/node_modules/
```
Comment (feross) [inline .gitignore:1]: Please remove this file. I put this into the global `.gitignore`.

File: README.md [hunk 2]
```diff
@@ -1,14 +1,10 @@
...
-[![downloads](https://img.shields.io/npm/dm/parse-torrent-file.svg)](https://npmjs.org/package/parse-torrent-file)
 
-#### Parse a .torrent file and return an object of keys/values
+[![travis](https://img.shields.io/travis/feross/parse-torrent-file/master.svg)](https://travis-ci.org/feross/parse-torrent-file) [![npm](https://img.shields.io/npm/v/parse-torrent-file.svg)](https://npmjs.org/package/parse-torrent-file) [![downloads](https://img.shields.io/npm/dm/parse-torrent-file.svg)](https://npmjs.org/package/parse-torrent-file)
 
 [![Sauce Test Status](https://saucelabs.com/browser-matrix/parse-torrent-file.svg)](https://saucelabs.com/u/parse-torrent-file)
 
-Works in node and the browser (with [browserify](http://browserify.org/)). This module is
-used by [WebTorrent](http://webtorrent.io)!
+Parse a .torrent file and return an object of keys/values. Works in node and the browser (with [browserify](http://browserify.org/)). This module is used by [WebTorrent](http://webtorrent.io)!
```
Comment (feross) [inline README.md:7]: Please undo these readme changes.

File: bin/index.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+var ArgumentParser = require('argparse').ArgumentParser
```
Comment (feross) [inline bin/index.js:1]: I prefer `minimist` for argument parsing.

Comment (feross) [inline bin/index.js:1]: No need to rename this file.

**Comment by feross** [comment]
- remove redundant fields & copies of buffers left in the parsed object

These are left intentionally for convenience and they're used throughout the webtorrent codebase. They're also present in https://github.com/feross/magnet-uri and exposed by `parse-torrent`. These fields do not prevent them from being converted to JSON and you're free to delete them before saving them to your DB.

``` js
    delete parsedTorrent.info
    delete parsedTorrent.infoBuffer
    delete parsedTorrent.infoHashBuffer
```
- improve the cli so it can do encoding and decoding via stdio (without needing a filesystem)

This is a good idea. Thanks for doing this.
- add support for some extra fields like encoding & md5sum

I don't see any code for this. Also, I don't know what this means.
- ensure that information isn't lost when converting to/from JSON (especially anything that would change the infohash)

Can you give an example of this happening? I've never had this issue.

**Comment by feross** [comment]
Actually stdin is already supported. You just need to install the `parse-torrent` package.

```
npm install parse-torrent -g
```

``` bash
cat file.torretn | parse-torrent --stdin
```

I'll remove the command line program from this package as it's redundant and maintain two largely similar command line programs isn't fun.

**Comment by feross** [comment]
Thanks for the PR, but I don't think we need these changes. Hope you understand.

**Comment by notslang** [comment]
> - add support for some extra fields like encoding & md5sum
> 
> I don't see any code for this. Also, I don't know what this means.

`encoding` and `md5sum` are optional fields in the torrent file spec... https://wiki.theory.org/BitTorrentSpecification#Metainfo_File_Structure talks about them a bit. And I haven't committed any of that code yet.

> - ensure that information isn't lost when converting to/from JSON (especially anything that would change the infohash)
> 
> Can you give an example of this happening? I've never had this issue.

There's some obvious info lost from the `encoding`, `md5sum`, and `announce-list` fields, but I'll let you know if I can find a example of something changing the infohash.

> Thanks for the PR, but I don't think we need these changes. Hope you understand.

No worries - I kinda expected that given how many changes are involved. I'll just maintain a fork.

**Comment by feross** [comment]
> There's some obvious info lost from the encoding, md5sum, and announce-list fields, but I'll let you know if I can find a example of something changing the infohash.

Fair point. `encoding` and `md5sum` seem like a bad idea to me and they're not very useful. No one seems to use `encoding` since lots of clients would fail if a non-standard encoding was used. And no BitTorrent client uses `md5sum`. So I'd rather just keep this implementation clean and not support them.

You have more of a point with `announce-list` since info is lost. But the announce list is outside of the `info` section, so it does not affect the info hash.

**Pull request closed without merging**